### PR TITLE
docs: consolidate API docs and update readme files

### PR DESCRIPTION
**Pull request by achingbrain**
Moves api docs into `/docs` folder similarly to how js-libp2p has organised their docs.

Tries to trim readmes down a bit to make them more digestible and adds links to futher information on different topics.

Also reinstates dependency statuses.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,122 +22,64 @@
...
-- [Contribute](#contribute)
-  - [Want to hack on IPFS?](#want-to-hack-on-ipfs)
+- [Want to hack on IPFS?](#want-to-hack-on-ipfs)
 - [License](#license)
 
+## Documentation
+
+* [Core API](./docs/core-api)
+* [Examples](./examples)
+* [Development]('./docs/DEVELOPMENT.md)
```
Comment (hugomrdias) [inline README.md]:
```suggestion
* [Development](./docs/DEVELOPMENT.md)
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- [Development documentation](#development-documentation)
+
+## API Docs
+
+`ipfs` can run as part of your program (an in-process node) or as a standalone daemon process that can be communicated with via an HTTP RPC API using the [`ipfs-http-client`](../packages/ipfs-http-api) module.
+
+Whether accessed directly or over HTTP, both methods support the full [Core API](#core-api).  In addition other methods are available to construct instances of each module, etc.
+
+* [Core API docs](./core-api/README.md)
+* [IPFS API]('../packages/ipfs/README.md)
```
Comment (hugomrdias) [inline docs/README.md]:
```suggestion
* [IPFS API](../packages/ipfs/README.md)
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+* [IPFS-HTTP-CLIENT API]('../packages/ipfs-http-client/README.md)
```
Comment (hugomrdias) [inline docs/README.md]:
```suggestion
* [IPFS-HTTP-CLIENT API](../packages/ipfs-http-client/README.md)
```

File: packages/interface-ipfs-core/README.md [hunk 4]
```diff
@@ -53,16 +62,16 @@ Include this badge in your readme if you make a new module that implements inter
 
 In JavaScript land:
 
-```js
-npm install interface-ipfs-core
+```console
```
Comment (hugomrdias) [inline packages/interface-ipfs-core/README.md]:
```suggestion
```bash
```

File: packages/interface-ipfs-core/README.md [hunk 5]
```diff
@@ -53,16 +62,16 @@ Include this badge in your readme if you make a new module that implements inter
...
+$ npm install interface-ipfs-core
 ```
 
 If you want to run these tests against a go-ipfs daemon, checkout [ipfs-http-client](https://github.com/ipfs/js-ipfs-http-client) and run test tests:
 
-```
-git clone https://github.com/ipfs/js-ipfs-http-client
-npm install
-npm test
+```console
```
Comment (hugomrdias) [inline packages/interface-ipfs-core/README.md]:
```suggestion
```bash
```

File: packages/interface-ipfs-core/README.md (hunk 4 above)
Comment (achingbrain) [reply packages/interface-ipfs-core/README.md]: Bash does bash script syntax highlighting which can result in weird output when you're trying to format console commands.

Comment (achingbrain) [reply packages/interface-ipfs-core/README.md]: bash:

```bash
$ npm install interface-ipfs-core
...some output
$ git clone https://github.com/ipfs/js-ipfs-http-client
$ npm install
$ npm test
```

console (similar but the command output is highlighted):

```console
$ npm install interface-ipfs-core
...some output
$ git clone https://github.com/ipfs/js-ipfs-http-client
$ npm install
$ npm test
```

Comment (hugomrdias) [reply packages/interface-ipfs-core/README.md]: ah cool didn't know

**Pull request merged**